### DISCUSSION TITLE
CSV export for Koinly with all Harmony network transactions?

### DISCUSSION THREAD

**Discussion by CoffeeIntheMorning** [pwillworth/dfkreport#7] [Ideas]
Hi, it looks like the new csv export for koinly will be really helpful for transactions made on DFK. Since Koinly has not yet integrated all Harmony transactions automatically, would it be possible to go one step further and provide an option on the dfktracker app to include all of a wallet's transactions on the Harmony network? This would be super helpful and a service i would love to donate for. :) 

Granted, having the export for just DFK transactions is helpful, but when the day comes when Koinly integrates all Harmony transactions, I'll have to go through and manually delete any DFK-related transactions it pulls in if I originally uploaded dfktracker's csv with only DFK transactions, if that makes  sense.

Thanks for considering!

**Comment by pwillworth** [comment]
What kind of transactions are you looking for that do not show up?  dfkreport currently looks at all Harmony activity, but may not be aware of or know how to interpret results of all smart contract transactions, so some stuff might not get represented.  It might be really easy to make it aware of simple DEX activity on other sites for example, but interpreting activity in other games on Harmony might be a whole other project in itself.  So, maybe if you share what you would like to see included, I can assess that.

**Reply by CoffeeIntheMorning** [reply]
Hi, thanks for responding! Basically, I'm looking for something I can import to Koinly that will at least show something for all Harmony transactions for a given wallet. The current blockchain explorer is difficult to understand and most platforms like Koinly haven't yet fully integrated Harmony, so there's a huge gap in the market here for tax reporting purposes - seems like DFKTracker is really close to filling that gap so thought I'd ask to see if it's possible!

Here are examples of what I'm not seeing in the export (not interested in other Harmony games, I just really want all bridging transactions, swaps, liquidity provision on all Harmony exchanges to show up):

-Liquidity in/out, staking LP tokens on DFK and other exchanges
-All transactions before September 9, 2021 (am i missing something? if this is when the jewel price feed was available, i would be happy to at least have an export where i can manually enter the price)
-Bridging assets in/out from (e.g., from Horizon bridge, Multichain, Synapse, etc.)
-Tranquil Finance lending/borrowing/TRANQ/xTRANQ staking
-harmony.curve.fi transactions
-Viper transactions and related assets (e.g., WAGMI, VIPER)
-Sushiswap transactions
-farmersonly.farm transactions - for that matter, recognizing contract interactions with everything on vfat.tools/harmony would be great

Thanks, sorry for the laundry list but figured I'd ask if it's doable.

Edit: it's likely too much to get the above perfect. Just showing an entry for every harmony transaction in the csv export, no matter how barebones, would be helpful so we can fill in the blanks. I think as long as it showed the tokens in the transaction, tx hash, and maybe what kind of transaction it was, we could fill in the price and add other details the app is not capable of identifying.

**Reply by pwillworth** [reply]
First, you keep on mentioning DFKTracker, so I want to make sure you're in the right place.  You know this repo is for the Lila's Ledger/dfkreport.cognifact.com NOT dfktracker.app right?

Assuming you're in the right place, your list is helpful, and some of it is already available and will look into filling the gaps, might not be a big lift.

**Reply by CoffeeIntheMorning** [reply]
Oh my, I am embarrassed. I apologize. Thank you for straightening me out. I found this  through a search for "koinly export" on the dfk discord and got the sites mixed up/out of context in the search results. I see Lila's Ledger now has a koinly export option too (thank you!). 

I tried the CSV export at https://dfkreport.cognifact.com/ and I'm impressed. It really does include a lot more! My wish list from a quick glance to get something that would account for all harmony transactions is:

-Liquidity in to LP tokens and out LP tokens, staking LP tokens on DFK and other exchanges (I could only find what I think are garden LP token withdrawals in the csv)
-Transactions before DFK's launch date
-Bridging assets in/out (e.g., from Horizon bridge, Multichain, Synapse, etc.)
-Transactions on other dexes (viper, curve, etc, others on vfat.tools/harmony
-Tranquil lending/borrowing/staking
-Option to filter for only Harmony or only Avalanche transactions

Thanks, and apologies in advance for missing anything above that's already there, and again for the confusion.

**Reply by pwillworth** [reply]
No worries!, easy to mix up.  Thanks for the feedback and taking the time to list out these things, I'll refer to them when I start to fill in some of these gaps.

**Reply by ryanmoore1029** [reply]
+1 on this.. the export is great so far, 

I also only see Liquidity OUT for some reason for DFK
I am missing Tranq / RVRS / MIS token swaps and staking (Tranq/RVRS IDO's on the Artemis platform and MIS is the native token for Artemis: https://app.artemisprotocol.one/)
+1 on Farmersonly liquidity in/out 

I would definitely be happy to contribute some $ as well.. thanks for your hard work.

**Comment by pwillworth** [comment]
@CoffeeIntheMorning FYI, today was published the option to exclude avalanche transactions (it is off by default now)

@ryanmoore1029 On the Tax Report, only liquidity out will show up there to capture the gains.  If you look on transactions tab in liquidity section, do you see any "deposit" events?

**Comment by CoffeeIntheMorning** [comment]
Hi @pwillworth great, thanks for adding that feature!

Yes, I see liquidity in in the liquidity section of the Summary tab - that's helpful. I mentioned that it would be ideal if the koinly export could include liquidity in transactions and LP staking deposits because Koinly encourages taking the conservative approach of treating such actions as taxable events themselves, but knowing nothing will never please everyone, this liquidity section in the summary definitely gives me something to work with (e..g, I could probably just insert one manual transaction in Koinly somewhere with the entire liquidity-in amount at some point before my first garden withdrawal for a given liquidity pair).

Thanks so much, I can't tell you how much I appreciate what you're creating here given how little attention Koinly has given to defi.

**Reply by pwillworth** [reply]
If you switch to the "Transactions" tab and then do the Export CSV, it will provide output that I think will be better for Koinly.  The csv of the tax report does some calculations that I think Koinly is wanting to do for you, so I think if you export the transaction level data instead, which include the "IN" events, that might work better for you.

**Comment by pwillworth** [comment]
FYI, Update went in today to capture Tranquil lending platform and staking stuff.  Unfortunately I haven't been able to accurately capture interest income/expenses for these transactions.  The receipt log data is really weird and limited.  Everything is tracked via lending pool tokens in the background and the only info emitted is how much change happens to those background tokens across the whole platform.  Interest gains/expenses may be inferrable by looking at the Summary tab info I added for lending though if viewing a date range that encompasses a clean entry and exit of the market if that makes sense.

**Comment by CoffeeIntheMorning** [comment]
Thanks, I noticed the export from the Transactions tab now has basically everything I’ve done in harmony and it gives me enough to work with for Tranq lending. I’m going through some csv manipulations as we speak! I did have to split the lp-in and lp-out transactions as koinly does not recognize the token amounts and token tickers separated by the “/“. No big deal though. I’ve been assigning null tokens to the dfk quest items and some tokens that aren’t recognized by koinly as well. It’s importing like a dream now. Thank you so much, I was freaking out that koinly still hadn’t integrated harmony and this is a huge help.

**Reply by pwillworth** [reply]
Nice!  That's great to hear.  Thanks for the feedback about the LP in/out, that will be pretty easy to make that show up on two lines for koinly.  Yeah, I figure there a whole bunch of weird game stuff that Koinly is going to have no idea about :-P

**Comment by CoffeeIntheMorning** [comment]
One thing, feel free to ignore if it’s a pain. Is it possible to have an option to show the fee (gas) of each transaction with ONE as the currency instead of USD? The USD value is of course great but would make ONE accounting easier if it was listed as gas currency. Thanks!

**Reply by pwillworth** [reply]
For this are you talking about in the koinly csv export to show one amount and ONE currency type for the Transaction fee?  If so, is that just in general what Koinly expects, or just something you are interested in seeing?

**Reply by CoffeeIntheMorning** [reply]
Yup, I mean it might be helpful for Koinly purposes to have the "Fee Amount" show the number of ONE spent on gas and the "Fee Currency" be ONE.  Koinly treats gas spent as a taxable event and keeps a running total for each token in your portfolio.

More detail for context, feel free to skip:
It's not a big to leave the csv how it is because Koinly is able to automatically import all ONE transactions (just no HRC-20 tokens because I think koinly devs are used to working with etherscan-like explorers), so it is already accounting for my ONE gas expenditures separately and I can zero out the Fee Amount and Fee Currency columns in Lila's Ledger's CSV. The only problem with leaving Koinly's automatic ONE imports in is that I'll have some ONE inflows and outflows already accounted for in transactions imported from Lila's CSV, but i can soft-delete automatically imported duplicates. Having the ONE totals in the fee column of the csv would allow me to do away with Koinly's automatically imported ONE transactions altogether, I think. Thanks!
